scripts/lib/ExtensionNameValidator.js

```diff
@@ -1,5 +1,8 @@
 /** @param {string} extensionName */
 exports.isValidExtensionName = (extensionName) => {
+  // Ensure extensionName is a string to prevent runtime errors
+  if (typeof extensionName !== 'string') return false;
+
```

Comment:
Typechecking is already statically enforced via typescript: this is unnecessary complexity.